### PR TITLE
feat(data drawer sidebar): Add full "drawer" support to Sidebar component #721

### PR DESCRIPTION
**Pull request by goswamianshuman**
the feature was required to be added in an issue with the title `Add full "drawer" support to Sidebar component #721`

"enhance #721"

## Description

this action contains some features which resolve the following: 

- Support for a sidebar that stashes itself (data-drawer style) at a certain min-width and displays a hamburger menu button 
instead

- Clicking the hamburger menu button re-displays the sidebar and puts an opaque overlay (drawer-backdrop) over the remaining screen
- Clicking a menu item in the sidebar performs the menu item's function (ie loads a new page) and re-stashes away the sidebar
- Clicking the opaque overlay re-stashes away the sidebar

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change contains a documentation update

## Breaking changes

Please document the breaking changes if suitable.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rluders — CHANGES_REQUESTED** [review]
Hey, @goswamianshuman thanks... But, TBH, I'm not even sure what I'm looking at. Why did you duplicate the sidebar component? Is there any particular reason to create a new component that duplicates an existing component? I don't think so... Am I missing something here?

File: src/lib/components/Modal/Modal.spec.tsx [hunk 1]
```diff
@@ -8,21 +8,21 @@ import type { ModalProps } from './Modal';
 import { Modal } from './Modal';
 
 describe('Components / Modal', () => {
-  it('should automatically focus the `TextInput` inside the `Modal` when its opened', async () => {
-    const root = document.createElement('div');
-    const user = userEvent.setup();
+  // it('should automatically focus the `TextInput` inside the `Modal` when its opened', async () => {
```
[the remark this answers is not in the document]
Comment (goswamianshuman) [reply src/lib/components/Modal/Modal.spec.tsx:11]: > But... why?

let me check it once I think I missed something here.

[the remark this answers is not in the document]
Comment (goswamianshuman) [reply src/lib/components/Modal/Modal.spec.tsx]: Hello, @rluders I just changed the changes in modal.spec.tsx back to the original in the local system there were getting some errors while building but I tried it again and they worked without any problem I think you can further check the other things and let me know if there is a need to change anything else. 

also, let me know about the above explanation for updating and grouping the elements into a new folder will this work? if not then I can change the names.

[the remark this answers is not in the document]
Comment (rluders) [reply src/lib/components/Modal/Modal.spec.tsx]: Well, I'm not convinced about this one. I think that the best way to make the `Sidebar` works as a drawer is to simply use the implementation that is in WIP (#664), and simply use the `Sidebar` as the `Drawer`content.

Currently, this is just a duplication of code that we already have... Doesn't make sense, IMHO. So, for me, it doesn't work.

@tulup-conner WDYT?

**Comment by tulup-conner** [comment]
Thank you for the submission, but it needs a lot more work. 

We also just totally revamped our development server, and you may want to read the [new contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md). A few things:

1. Since we just changed almost everything in the development server, you need to rebase this branch with the latest changes. Let me know if you can't do that, and I'll do so for you
2. You need to add a new page in the docs with the exact same examples found at https://flowbite.com/docs/components/drawer/
3. You can't duplicate all of the code from Sidebar to accomplish this. You should just create a `Sidebar` internally with `Drawer` and customize it with a theme to look like the Flowbite `Drawer`
5. The component should be in a directory just called `src/components/drawer`
6. There should be unit tests specific only to the `Drawer`, because what you're effectively doing is running all of the `Sidebar` unit tests twice for no reason. That would be things like making sure the body can't be scrolled when the `Drawer` is open

**Pull request closed without merging**